### ISSUE TITLE
aws_route53_resolver_rule domain_name value "." is now broken

### ISSUE THREAD

**Issue by mingfang** [terraform-providers/terraform-provider-aws#14730]
aws_route53_resolver_rule domain_name with value "." is legal and essential.
It was working until this commit.

https://github.com/terraform-providers/terraform-provider-aws/commit/ee459d6e05e6c5a68daa7becbe5fafb31c1345f4#diff-c728f6192d56e060a6fa9964d8d90832

that removed `suppressRoute53ZoneNameWithTrailingDot()` from `aws/diff_suppress_funcs.go`
This is a critical problem for me with no workarounds besides manually creating the rule.

**Comment by ewbankkit** [comment]
Being able to create a rule for the `.` domain name is explicitly called out in the [AWS documentation](https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/resolver.html#resolver-overview-forward-vpc-to-network-using-rules):

> If you want to forward all queries to your network, you create a rule, specify "." (dot) for the domain name, and associate the rule with the VPCs for which you want to forward all DNS queries to your network.

**Comment by ewbankkit** [comment]
Attempting to create a rule with `domain_name = "."` results in the error:

```
        Error: error creating Route 53 Resolver rule: InvalidParameter: 1 validation error(s) found.
        - minimum field size of 1, CreateResolverRuleInput.DomainName.
```

**Comment by bflad** [comment]
The fix for this has been merged and will release with version 3.5.0 of the Terraform AWS Provider, very shortly. Thanks to @ewbankkit for the implementation. 👍
